internal/mining/mining.go

```diff
@@ -931,6 +931,9 @@ func NewBlkTmplGenerator(cfg *Config) *BlkTmplGenerator {
 // transaction and its ancestors into account.
 func calcFeePerKb(txDesc *TxDesc, ancestorStats *TxAncestorStats) float64 {
 	txSize := txDesc.Tx.MsgTx().SerializeSize()
+	if ancestorStats.Fees < 0 || ancestorStats.SizeBytes < 0 {
```

Comment:
> This change modifies the block template generator to account for some ancestor statistics not being updated correctly during a reorg

Is the plan to fix the ancestor statistics calculation during a reorg, and this is just a temporary stop-gap?  We should probably add a comment here for what this case is handling.

If these can be a negative value, do you think we should make updates [here](https://github.com/decred/dcrd/blob/d1e657f71c639533f71aea37beb8e45c4b91208c/internal/mining/mining.go#L1252) and [here](https://github.com/decred/dcrd/blob/d1e657f71c639533f71aea37beb8e45c4b91208c/internal/mining/mining.go#L1451) to call that out and/or handle that case explicitly?

Reply:
> Is the plan to fix the ancestor statistics calculation during a reorg, and this is just a temporary stop-gap?

Yes, the plan is to rework how transactions are handled during reorgs to avoid holes in the tx graph, though I think these checks should be left in-place until there are tests that guarantee this path is not possible.

> If these can be a negative value, do you think we should make updates here [1] and here [2] to call that out and/or handle that case explicitly?

[1] - I do not think any updates need to be made here in this PR, but this field `TxDesc.fee` can be safely removed since it's not read from at this point. I did not remove it to keep the scope of this PR focused.
[2] - No, I think the cache should be accurate at this point since the call to `ancestors` updates it. This behavior should not affect the loop that fills the block template outside of prioritization.

Reply:
That makes sense.  Thanks for clarifying!